great_expectations/expectations/core/expect_table_columns_to_match_ordered_list.py

```diff
@@ -330,3 +336,54 @@ def _validate(
                     "details": {"mismatched": mismatched},
                 },
             }
+
+    @classmethod
+    @renderer(renderer_type=AtomicDiagnosticRendererType.OBSERVED_VALUE)
+    def _atomic_diagnostic_observed_value(
+        cls,
+        configuration: Optional[ExpectationConfiguration] = None,
+        result: Optional[ExpectationValidationResult] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> RenderedAtomicContent:
+        renderer_configuration = RendererConfiguration(
+            configuration=configuration,
+            result=result,
+            runtime_configuration=runtime_configuration,
+        )
+        ov_param_prefix = "ov__"
+        ov_param_name = "observed_value"
+
+        renderer_configuration.add_param(name="column_list", param_type=RendererValueType.ARRAY)
+        renderer_configuration.add_param(
+            name=ov_param_name,
+            param_type=RendererValueType.ARRAY,
+            value=result.result.get("observed_value"),
+        )
+        renderer_configuration.add_param(
+            name="observed_value_label", param_type=RendererValueType.STRING, value="Table columns"
+        )
+        renderer_configuration = cls._add_array_params(
+            array_param_name=ov_param_name,
+            param_prefix=ov_param_prefix,
+            renderer_configuration=renderer_configuration,
+        )
+        observed_value_str: str = cls._get_array_string(
+            array_param_name=ov_param_name,
+            param_prefix=ov_param_prefix,
+            renderer_configuration=renderer_configuration,
+        )
+        renderer_configuration.template_str = f"$observed_value_label: {observed_value_str}"
```

Comment:
There is an opportunity to refactor the lines before and after this up into the `Expectation` class (see jira ticket). Did you decide not to do that or did you not see that part?